src/code/z_map_exp.c

```diff
@@ -1,31 +1,231 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_exp/func_8010A000.s")
+static s16 sPlayerInitialPosX = 0;
+static s16 sPlayerInitialPosZ = 0;
+static s16 sPlayerInitialDirection = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_exp/func_8010A074.s")
+static s32 sDungeonAndBossScenes[] = {
+    SCENE_MITURIN,    SCENE_HAKUGIN,    SCENE_SEA,    SCENE_INISIE_N,  SCENE_INISIE_R,
+    SCENE_MITURIN_BS, SCENE_HAKUGIN_BS, SCENE_SEA_BS, SCENE_INISIE_BS,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_exp/func_8010A0A4.s")
+static s32 sDungeonScenes[] = {
+    SCENE_MITURIN, SCENE_HAKUGIN, SCENE_SEA, SCENE_INISIE_N, SCENE_INISIE_R,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_exp/func_8010A0F0.s")
+static s32 sBossScenes[] = {
+    SCENE_MITURIN_BS,
+    SCENE_HAKUGIN_BS,
+    SCENE_SEA_BS,
+    SCENE_INISIE_BS,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_exp/func_8010A164.s")
+static s32 D_801BF5A4[] = {
+    SCENE_22DEKUCITY,
+    SCENE_KOEPONARACE,
+    SCENE_F01,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_exp/func_8010A194.s")
+static s16 sLastRoomNum = 99;
```

Comment:
I'm a bit confused by that last sentence. Do you mean "move them inside the functions"? Because they don't match in the functions unless they're `static`. Or do you mean "remove the `static` from them, and then also move them around so they're closer to the functions that use them"?